pallets/creditcoin/src/types/loan_terms.rs

```diff
@@ -7,27 +7,26 @@ use scale_info::TypeInfo;
 use sp_runtime::traits::UniqueSaturatedInto;
 use sp_std::convert::TryFrom;
 
-pub type InterestRate = u64;
+pub type RatePerPeriod = u64;
+pub type Decimals = u64;
+pub type Period = u64;
 
-pub const INTEREST_RATE_PRECISION: u64 = 10_000;
+#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub struct InterestRate {
+	pub rate_per_period: RatePerPeriod,
+	pub decimals: Decimals,
+	pub period_ms: Period,
+}
 
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
 pub struct LoanTerms<Moment> {
 	pub amount: ExternalAmount,
 	pub interest_rate: InterestRate,
-	pub maturity: Moment,
-}
-
-fn calc_interest(principal_amount: &ExternalAmount, interest_rate: InterestRate) -> ExternalAmount {
-	principal_amount * interest_rate / INTEREST_RATE_PRECISION
-}
-
-impl<Moment> LoanTerms<Moment> {
-	pub fn calc_interest(&self) -> ExternalAmount {
-		calc_interest(&self.amount, self.interest_rate)
-	}
+	pub term_length_ms: Moment,
```

Comment:
Why do we use the generic `Moment` here? Also, `Moment` should probably be `Duration` or something since it no longer represents a moment in time but instead a `Duration` of time. Maybe `Period` should be `Duration`.